libkbfs/backpressure_disk_limiter.go

```diff
@@ -469,6 +480,52 @@ func (bdl *backpressureDiskLimiter) onBlocksDelete(
 	bdl.fileTracker.onBlocksDelete(blockFiles)
 }
 
+func (bdl *backpressureDiskLimiter) onDiskBlockCacheDelete(
+	ctx context.Context, blockBytes int64) {
+	if blockBytes == 0 {
+		return
+	}
+	bdl.lock.Lock()
+	defer bdl.lock.Unlock()
+	bdl.diskCacheByteTracker.onBlocksDelete(blockBytes)
+}
+
+func (bdl *backpressureDiskLimiter) beforeDiskBlockCachePut(
+	ctx context.Context, blockBytes int64) (
+	availableBytes int64, err error) {
+	if blockBytes == 0 {
+		// Better to return an error than to panic in ForceAcquire.
+		return 0, errors.New("backpressureDiskLimiter.beforeDiskBlockCachePut" +
+			" called with 0 blockBytes")
+	}
+	bdl.lock.Lock()
+	defer bdl.lock.Unlock()
+	freeBytes, _, err := bdl.freeBytesAndFilesFn()
+	if err != nil {
+		return 0, err
+	}
+
+	bt := bdl.diskCacheByteTracker
+	bt.updateFree(freeBytes)
```

Comment:
So all byte trackers need to keep track of the space used by both the journal and the cache. Otherwise, if each tracker independently thinks it's the only one using the free space, then they'll affect each other implicitly.

Example: Say you have a 100 GB limit and 25% limits for both the journal and cache, both which start off empty. Both trackers see the usage fraction as 0/(0 + 100), so let's say both fill up fairly quickly to 25GB each. But then, on the next journal block put, the journal sees 50 GB free and knows it's using 25 GB, so it calculates its usage fraction as 25/(25 + 50) = 33%, which is over the limit, and thus starts applying backpressure, which is undesirable, since by having separate limits we wanted to avoid having to deal with caching affecting journaling and vice versa.

The cleanest solution I can think of is to treat 'used' journal bytes as 'free' in the cache byte tracker, and to treat 'used' cache bytes as 'free' in the journal byte tracker. For example, here, it would look like:

```
bt.updateFree(freeBytes + bdl.journalByteTracker.used)
```

I think it suffices to update the free bytes only when we'd normally do it, and not every time the `used` field changes.